tools/dev/src/main/groovy/intellijRunConfig.groovy

```diff
@@ -57,13 +58,22 @@ Map controllerEnv = null
 Map invokerEnv = null
 
 containerNames.each{cn ->
+
     //Inspect the specific container
     def inspectResult = "docker inspect $cn".execute().text
     def json = new JsonSlurper().parseText(inspectResult)
 
     def imageName = json[0].'Config'.'Image'
     if (imageName.contains("controller") || imageName.contains("invoker")){
-        def mappedPort = json.'NetworkSettings'.'Ports'.'8080/tcp'[0][0].'HostPort'
+        // extract all the ports exposed by this container
+        def hostPorts = []
+        json[0].'NetworkSettings'.'Ports'.each{ port, portValue ->
+            if (portValue != null) {
+                hostPorts << portValue[0].'HostPort'
+            }
+        }
+        // look for the first port with [HostIp,HostPort] in range (8000,9000)
+        def mappedPort = hostPorts.find{ it.toInteger() > 8000 && it.toInteger() < 9000 }
```

Comment:
using OW ansible to run locally, controller runs on 10001, invoker on 12001 - my PORT env config ended up empty, since it is outside of this range; I would be ok with picking a hard coded port and using that (i.e. remove the PORT config from containers env map, replace it with a hard coded one). Otherwise, maybe just leave the config alone  - I'm not sure the need for replacing the container configured port, if that is the intention? (I assume it is ok to require stopping the container to run the host jvm etc).

Reply:
thanks for pointing this out @tysonnorris. Picking a pre-defined port for controller and another one for invoker is an easier approach, instead of guessing ports.

Reply:
done. the groovy scripts picks `10001` for controller and `12001` for invoker. docs are updated too.